src/math/Interpolate.cc

```diff
@@ -157,7 +157,7 @@ ::gsl_interp_type const *styleToGslInterpType(Interpolate::Style const style) {
         case Interpolate::UNKNOWN:
             throw LSST_EXCEPT(pex::exceptions::InvalidParameterError,
                               "I am unable to make an interpolator of type UNKNOWN");
-        case Interpolate::NUM_STYLES:
+        default:
```

Comment:
Russell's suggestion relies on `#pragma GCC diagnostic`, which is broken at the moment. It would certainly be my preferred solution otherwise. :disappointed: